khatm

ARHariri opened the pull request.

ARHariri committed:
1- Add capability to start/stop khatm
2- Add capability to store allowed request in storage when network is disconnected.
3- Add capability to commit read/unread pages
4- Change loading data strategy (Load data when user login, refresh or back to app) (Also need reconciliation local data in app with data in server)
5- Add some basic structure for commitments form

ARHariri committed:
1- Merge with master branch

ARHariri committed:
1- Change color to right menu and khatm form in night/day mode

ARHariri committed:
1- Fixing some bugs

ARHariri committed:
1- Fixing some bugs when user commit pages

ARHariri committed:
1- Add capability to commit read pages manually (select individually page number or select range of page numbers)
2- remove remainDays (unused variable in khatm form)
3- Improve rest days value (Fix bugs in interval between start and end dates)
4- Remove commit pages from storage
5- Set return value in http service when network is disconnect

aminazar commented on src/app/app.module.ts:
```diff
@@ -31,6 +31,8 @@ import {CreateKhatmPage} from "../pages/create-khatm/create-khatm";
 import {LanguageService} from "../services/language";
 import {KhatmService} from "../services/khatm.service";
 import {SocialSharing} from "@ionic-native/social-sharing";
+import {Network} from "@ionic-native/network";
```
do we use this import in this file?

ARHariri replied:
Yes. When remove it from the code, we have an error: 'No provider for network'

aminazar commented on src/components/left-menu/left-menu.html:
```diff
@@ -17,7 +17,7 @@
       </ion-item>
     <button ion-item detail-none (click)="nightMode()">
-      <ion-icon [name]="stylingService.nightMode?'eye':'eye-off'" item-left></ion-icon>
+      <ion-icon [name]="stylingService.nightMode ? 'sunny' : 'moon'" item-left></ion-icon>
```
good replacement :)

aminazar commented on src/components/right-menu/right-menu.html:
```diff
@@ -1,21 +1,35 @@
         </ion-col>
         <ion-col col-6>
-          <button ion-button menuClose *ngIf="isLoggedIn" block (click)="openPage('khatm')">{{ls.translate('Create Khatm')}}</button>
+          <button [color]="conditionalColoring.primary" ion-button menuClose *ngIf="isLoggedIn" block (click)="openPage('khatm')">{{ls.translate('Create Khatm')}}</button>
```
great - thanks for fixing background colour issue for side menus

aminazar commented on src/services/language.ts:
```diff
@@ -140,7 +140,8 @@ export class LanguageService {
     if(this._lang === 'fa'){
-      momentJalali.loadPersian(true);
+      // momentJalali.loadPersian(true);
+      momentJalali.loadPersian();
```
it did not work without 'true' parameter for me or Ms Salehi

ARHariri replied:
It is strange. It did not work with 'true' parameter for me. For uploading on ionic server I should remove the 'true' parameter.

ARHariri replied:
Sorry for late. Resolve this issue. I remove 'jalali-moment' package and install it again. This function need one Boolean parameter.

aminazar reviewed (commented):
Thanks - it seems you finished commitment page.
Did you upload the app to ionic after that?

ARHariri committed:
1- Improve commitments pages backed code

ARHariri committed:
1- Fix bugs when user select end page then start page in commitment page
2- Fix bug in commtiPage function when some pages swap from read to unread
3- Fix bugs in concurrent actions on storage (add and delete commit pages)

ARHariri committed:
1- Fix bugs in loadPersian function with entry parameter. This function should accept one boolean parameter

ARHariri committed:
1- Improve the chagne khatm dates and duration strategy

ARHariri committed:
1- Change read pages when commite/uncommit some pages

ARHariri committed:
1- Update read pages when commit/cancel some pages
2- Disable commit page button when commit page numbers change until save the changes
3- Fix bug when user just press instead of tap on one page to commit (User can commit now)

ARHariri committed:
Disable reading/stop khatm and commit pages buttons when the commit page model change
Extract assigning value to isChangingCommitment from html code into function in commitments typescript file (Not working correctly in all system before this change)

ARHariri committed:
Add translations of words and phrases in commitments, khatms and some part in right menu and registration pages.
